sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java

```diff
@@ -189,16 +189,16 @@ public OpenAIClientBuilder credential(AzureKeyCredential azureKeyCredential) {
         return this;
     }
 
-    private NonAzureOpenAIKeyCredential nonAzureOpenAIKeyCredential;
+    private KeyCredential nonAzureOpenAIKeyCredential;
 
     /**
-     * The NonAzureOpenAiKeyCredential used for public OpenAi authentication.
+     * The KeyCredential used for public OpenAi authentication.
      *
-     * @param nonAzureOpenAIKeyCredential The credential for non-azure public OpenAI authenticaton.
+     * @param credential The credential for non-azure public OpenAI authenticaton.
      * @return the object itself.
      */
-    public OpenAIClientBuilder credential(NonAzureOpenAIKeyCredential nonAzureOpenAIKeyCredential) {
-        this.nonAzureOpenAIKeyCredential = nonAzureOpenAIKeyCredential;
+    public OpenAIClientBuilder credential(KeyCredential credential) {
```

Comment:
We should remove `credential(AzureKeyCredential)` overload. We now have one KeyCredential that works for both Azure and OpenAI endpoints. We will need additional changes in this library to determine which endpoint to use and can't rely on `instanceof` check anymore.